### ISSUE TITLE
Support for Spigot IP Forwarding

### ISSUE THREAD

**Issue by Katelyn-SN** [Poweruser/MinetickMod#12]
Add Support for bungeecord ip forwarding / uuid forwarding 

They created something to add it to vanilla

https://github.com/thinkofdeath/vanillacord

But i would like to see MineTick include this as MineTick servers do make very good lobby servers.

**Comment by Poweruser** [comment]
Yes, BungeeCord support by default was long on the todo-list already. I just was never really bothered as for banning we are using BungeeBan and for authentication we built ourselves a set of bungee/bukkit plugins that communicate through plugin channels, so we don't depend on that. But I guess most of you don't have that.
UUID forwarding is a must though from 1.7.9 on, as the linked servers themselves are running in offline-mode

**Comment by Katelyn-SN** [comment]
You need UUID forwarding otherwise everything breaks its kinda  must i was able to deal without it pre 1.7.9 by a few plugins but it doesn't work at all now.

**Comment by Poweruser** [comment]
I created a new branch for this feature: [v1.7.9_bungeeSupport](https://github.com/Poweruser/MinetickMod/tree/v1.7.9_bungeeSupport)
Now let's see if I have missed anything...
There are a couple of extra debug messages in it, that will be gone once it's tested and working

**Comment by Katelyn-SN** [comment]
I tested it on the latest BungeeCord and it seems to be working correctly.

(Not a live environment of course so there could be possible issues i did not catch)

**Comment by Poweruser** [comment]
Ok.
One more thing: Like I've wrote in the commit message, I don't change the API here.
Spigot's patches
- add a new constructor to the PlayerLoginEvent and a new method for getting the forwarded separately address. I decided to reuse its existing method getAddress(). Depending on the setup it returns either the usual address or the forwarded one.
- add a new method to the Player interface, which CraftPlayer does of course implement. I'm just adding this method to CraftPlayer, but not the interface

I decided very early on, that I don't want to change the API. The MinetickMod is supposed to consist of internal changes only.
